hbase-common/src/main/java/org/apache/hadoop/hbase/util/ByteBufferArray.java

```diff
@@ -102,8 +170,8 @@ public int getMultiple(long start, int len, byte[] dstArray, int dstOffset) {
 
   private final static Visitor GET_MULTIPLE_VISTOR = new Visitor() {
     @Override
-    public void visit(ByteBuffer bb, byte[] array, int arrayIdx, int len) {
-      bb.get(array, arrayIdx, len);
+    public void visit(ByteBuffer bb, int pos, byte[] array, int arrayIdx, int len) {
+      ByteBufferUtils.copyFromBufferToArray(array, bb, pos, arrayIdx, len);
```

Comment:
Why change this? Seems this not belong to HBASE-17738?

Reply:
> Why change this? Seems this not belong to HBASE-17738?

Also removed some Unnecessary locks related to [HBASE-15785](https://issues.apache.org/jira/browse/HBASE-15785)

Reply:
Should i backport this patch first?

Reply:
This backport have to depend on that patch? Should be another backport if it is necessary.